File: codechef/contests/MARCH16/MAXISUM.cpp
#include <iostream>
#include <cmath>

using namespace std;

long long interaction(long long a[], long long b[], long long n){
	long long ans = 0;
	for (long long i = 0; i < n; i += 1){
		ans += a[i]*b[i];
	}
	return ans;
}

long long solve(long long a[], long long b[], long long n, long long k){
	long long mn = b[0];
	long long mp = b[0];
	long long ii=0, jj=0;
	for (long long i = 0; i < n; i += 1){
		if (b[i] > mp){
			mp = b[i];
			ii = i;
		}
		if (b[i] < mn){
			mn = b[i];
			jj = i;
		}
	}
	if (abs(b[ii]) > abs(b[jj])){
		if (b[ii] > 0){
			a[ii] += k;
		}else{
			a[ii] -= k;
		}
	}else{
		if (b[jj] > 0){
			a[jj] += k;
		}else{
			a[jj] -= k;
		}
	}
	return interaction(a, b, n);
}

int main(void){

	long long t, n, k;

	cin >> t;
	while(t--){
		cin >> n >> k;
		long long a[n], b[n];
		for (long long i = 0; i < n; i += 1) cin >> a[i];
		for (long long i = 0; i < n; i += 1) cin >> b[i];
		cout << solve(a, b, n, k) << endl;
	}

}

File: codechef/contests/MARCH16/MAXISUM.py

def interaction(a, b):
	return sum([x*y for x,y in zip(a,b)])

def solve(a, b, k):
	mn = b[0]
	mp = b[0]
	for s in b:
		if s < mn:
			mn = s
		if s > mp:
			mp = s
	i = mn if abs(mn) > abs(mp) else mp
	i = b.index(i)
	if b[i] < 0:
		a[i] -= k
	else: a[i] += k
	return interaction(a, b)

I = lambda : map(int, raw_input().split())

for _ in xrange(input()):
	n, k = I()
	a = I()
	b = I()
	print solve(a, b, k)